### PR TITLE
Fix expired stored token

### PR DESCRIPTION
**Pull request by tsuyoshizawa**
When get expired stored token, it should be as new access token.

### CODE REVIEW

File: scala-oauth2-core/src/main/scala/scalaoauth2/provider/GrantHandler.scala [hunk 1]
```diff
@@ -31,8 +31,8 @@ trait GrantHandler {
       }.getOrElse {
         handler.createAccessToken(authInfo)
       }
-      case Some(token) => Future.successful(token)
-      case None => handler.createAccessToken(authInfo)
+      case Some(token) if !token.isExpired => Future.successful(token)
```
Comment (rmmeans) [inline scala-oauth2-core/src/main/scala/scalaoauth2/provider/GrantHandler.scala]: I'm not following why this is needed - what was the scenario that this was broken for?

Comment (tsuyoshizawa) [inline scala-oauth2-core/src/main/scala/scalaoauth2/provider/GrantHandler.scala]: A developer is using Implicit grant, which doesn't support refresh token. Then he always get access token even if the token is expired. He doesn't want to return the expired token to users. So, I try to fix the problem.

Shouldn't we support re-issue access token automatically?
User can check an expired token on `getStoredAccessToken` by using `token.isExpired`. 

@rmmeans What do you think?

Comment (rmmeans) [inline scala-oauth2-core/src/main/scala/scalaoauth2/provider/GrantHandler.scala]: The thing with implicit grant flows is they don't support refresh tokens for a security reason - that you can't trust a web browser with long lived tokens. So refreshing the token for them automatically is essentially doing the same thing as just giving them a refresh token. For implicit grant, there really is no way that I'm aware of besides requiring a user to get a new token by authenticating again every time the token expires, it is the side-effect of choosing to use this grant flow.

**Comment by tsuyoshizawa** [comment]
Thanks. I close this PR.

If user want to cached token on implicit grant, check and return `None` value on `getStoredAccessToken` method.

**Pull request closed without merging**